src/JobScheduler.java

```diff
@@ -0,0 +1,168 @@
+import java.util.Scanner;
+/*
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+Design a class JobScheduler to simulate FCFS (First Come First Serve) scheduling algorithm. 
+FCFS means the process which arrives first, gets executed first. 
+
+Assume that we are receiving a number of processes with their arrival time and burst time seconds in a two dimensional array as input.  For example:
+
+[0][10]
+[6][20]
+[60][10]
+[110][5]
+
+Define methods to perform following operations: 
+
+Calculate completion time for each process. 
+Calculate waiting time for each process.
+Calculate turn around time for each process.
+Average waiting time of processes.
+Maximum waiting time period for a process in queue.
+
+
+Here we have simple formulae for calculating various times for given processes:
+
+Completion Time: Time taken for the execution to complete, starting from arrival time of first process.
+Turn Around Time: Time taken to complete after arrival. In simple words, it is the difference between the Completion time and the Arrival time.
+Waiting Time: Total time the process has to wait before it's execution begins. It is the difference between the Turn Around time and the Burst time of the process.
+Burst Time : Time required to execute a process.
+
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+ */
+
+
+
+public class JobScheduler
+{
+	/* Process 
+	 * process[n][0]-arrival time
+	 * process[n][1]-burst time
+	 * process[n][2]-waiting time
+	 * process[n][3]-Completion time
+	 * process[n][4]-turn around time
+	*/
+	static int process[][];
+	/*
+	 * variable for computation of maximum waiting time and average turn around time
+	 */
+	static int process_count=0;
+	static int max_waiting=0;
+	static int total_waiting=0;
+	public static void main(String args[])
+	{
+		//fetchDetails(new int[][]{{0,10},{6,20},{60,10},{110,5}}); // Fetching process arrival and burst time
+		//fetchDetails(new int[][]{{0,5},{0,10},{1,5},{10,2},{20,10}});// Case 2
+		fetchDetails();
+		calculate(); // Calculating other parameters from fetched details
+		printDetails(); // Printing Details of various times related to process
+	}
+	
+	// Taking input about the process arrival time and process burst time
+	public static void fetchDetails(int arr[][])
+	{
+		process=new int[arr.length][5];
+		/*
+		 * For taking input in form of 2D array of size N X 2
+		 * 0 - arrival time
+		 * 1 - burst time
+		 */
+		for(int n=0;n<arr.length;n++)
+		{
+			process[n][0]=arr[n][0];
+			process[n][1]=arr[n][1];
+		}
+	}
```

Comment:
I don't think you need this method. You could initialize this process array at line 87.

Reply:
> I don't think you need this method. You could initialize this process array at line 87.

It's a helper function for fetch details 

- It is used to copy user input to process.
- It is also useful in case used provides the process details in form of 2D array of size NX2.